### PR TITLE
Working transformer lens integration with circuitviz demo for attention

### PR DESCRIPTION
**Pull request by Jazhyc**
Sorry, I was not aware that the previous pull request was already closed.

This pull request replaces the traditional nn.Module parent class used in the model components with HookedRootModule. I also added some hook points to variables that I thought would be of interest.

As a demonstration, I added an attention visualization graph from circuitsviz

### CODE REVIEW

File: model/attention.py [hunk 1]
```diff
@@ -2,14 +2,23 @@
...
         self.num_heads = config.num_heads
         self.hidden_size = config.hidden_size
         self.head_dim = config.hidden_size // config.num_heads
         self.config = config
+        
+        self.hook_k = HookPoint()
+        self.hook_v = HookPoint()
+        self.hook_q = HookPoint()
+        self.attn_hook = HookPoint()
+        self.context = HookPoint()
```
Comment (keenanpepper) [inline model/attention.py]: I think all the hooks should have "hook" in the name.

**Review by keenanpepper — COMMENTED** [review]
Please rename all the hook functions to look like "hook_something" and then merge this in.

I'm so glad you found such a simple way to do this that works; great job!

File: model/attention.py (hunk 1 above)
Comment (Jazhyc) [reply model/attention.py]: I agree, I still find naming variables quite difficult

**Comment by Jazhyc** [comment]
Thanks. It seems like there might be a bug with the train.py script which causes it to set the ablation masks correctly. After renaming the variables and fixing the configuration problem, I'll make a merge
